### PR TITLE
[OpaqueValues] Small fixes for AddressLowering.

### PR DESCRIPTION
**Pull request by nate-chandler**
- Allow deletion of alloc_stacks for which dealloc_stacks have been created.
- Handle the fix_lifetime instruction.

Based on https://github.com/apple/swift/pull/61893 .

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test macOS platform

File: lib/SILOptimizer/Utils/InstructionDeleter.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@ static bool hasOnlyIncidentalUses(SILInstruction *inst,
   for (SILValue result : inst->getResults()) {
     for (Operand *use : result->getUses()) {
       SILInstruction *user = use->getUser();
+      if (isa<DeallocStackInst>(user) && isa<AllocStackInst>(inst))
```
Comment (atrick) [inline lib/SILOptimizer/Utils/InstructionDeleter.cpp:26]: Deleting an `alloc_stack` is supposed to be handled by `isScopeAffectingInstructionDead`. We cannot remove `alloc_stack` at -Onone without checking varInfo

**Comment by nate-chandler** [comment]
@swift-ci please test

**Review by atrick — APPROVED** [review]
!

**Pull request merged**